Azure/azure-cli-extensions
# [AKS] `az aks create/update`: Support UserAssigned Managed Identity for grafana linking in managed prometheus

---

This checklist is used to make sure that common guidelines for a pull request are followed.

```az aks create -n kaveeshcli22 -g kaveeshcli --location westeurope --enable-azure-monitor-metrics --azure-monitor-workspace-resource-id "{full_id}" --grafana-resource-id "{full_id}"```

```az aks create -n kaveeshcli22 -g kaveeshcli --disable-azure-monitor-metrics ```


**Description**
Azure Managed Grafana now has support for UserAssigned Managed Identity and we're adding support for it in the managed prometheus addon.

**Testing Guide**
- Set context to one of the air gapped clouds
- Use the command with the grafana-resource-id parameter similar to `az aks create -n kaveeshcli22 -g kaveeshcli --location {cluster_location} --enable-azure-monitor-metrics --azure-monitor-workspace-resource-id "{full_id}" --grafana-resource-id "{full_id}"` and it should result in a a successful result. Please use a grafana resource that is creating/updated using a user assigned managed identity


### General Guidelines

- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:



### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.
